plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -604,6 +605,14 @@ private void invalidateTableStatisticsCache(String databaseName, String tableNam
                 .forEach(tableCache::invalidate);
     }
 
+    private void invalidateTablesWithParameterCache(String databaseName, String tableName)
+    {
+        tablesWithParameterCache.asMap().entrySet().stream()
+                .filter(cacheEntry -> cacheEntry.getKey().getDatabaseName().equals(databaseName) && cacheEntry.getValue().contains(tableName))
```

Comment:
iterate over keySet

Reply:
Sure. Updated to iterate over keySet and then do a lookup for each key.